api/v1beta1/cloudstackmachine_webhook.go

```diff
@@ -84,15 +84,15 @@ func (r *CloudStackMachine) ValidateUpdate(old runtime.Object) error {
 	}
 	oldSpec := oldMachine.Spec
 
-	errorList = webhookutil.EnsureBothFieldsAreEqual(r.Spec.Offering.ID, r.Spec.Offering.Name, oldSpec.Offering.ID, oldSpec.Offering.Name, "offering", errorList)
+	// errorList = webhookutil.EnsureBothFieldsAreEqual(r.Spec.Offering.ID, r.Spec.Offering.Name, oldSpec.Offering.ID, oldSpec.Offering.Name, "offering", errorList)
```

Comment:
@rejoshed, are you comfortable with removing these ValidateUpdate constraints?

Reply:
It's necessary as otherwise users can not upgrade the cluster which requires using a new template or scale up the nodes which requires a change in offering. I've tested the changes and they work without any issues. Planning to open a seperate PR just for this change detailing the need to do so